### PR TITLE
Superfast loaders

### PR DESCRIPTION
**Pull request by kyranet**
### Proposed Semver Increment Bump: [MAJOR]

Loading times, tested in two bots:
108 commands, (~10.200loc) 35s => 6s (5.8x faster)
78 commands, (~6.400loc) 24s => 5s (4.8x faster)

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Faster loading times
- Cache optimization

### Fixes:

- TypeError from `awaitMessage`, as in #158

### CODE REVIEW

**Comment by kyranet** [comment]
I added a last change, deleting the file from the cache after loading the pieces. With that, I could reduce ~5MB worth of data from the cache.

On my point of view, I don't think we need the same data cached in both `require.cache` and `client.CollectionOfPieces`. I also thought this change *may* break the reload command, but I have just tested and it doesn't return an error.

**Comment by kyranet** [comment]
According to the reload command. 

- Reload command: **OK!**
- Reload monitor: **OK!**
- Reload inhibitor: **OK!**
- Reload function: **OK!**

Memory usage seems to be better, dropped 8MB with all the previous PRs (now using 41MB vs 49MB). As well as the loading time is now faster since we removed a bunch of useless functions. ¯\\\_(ツ)\_/¯

**Comment by kyranet** [comment]
Lastest commit fixes the issue #158.

File: functions/awaitMessage.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const options = {
 };
 
 module.exports = async (client, msg, cmd, args, error) => {
-  const permLvl = await client.funcs.permissionLevel(client, msg.author, msg.guild).catch(err => client.emit("error", client.funcs.newError(err)));
+  const user = msg.author || msg.member;
```
Comment (CyberiumShadow) [inline functions/awaitMessage.js]: shouldn't it be `const user = msg.member || msg.author;` ?

**Review by CyberiumShadow — CHANGES_REQUESTED** [review]

File: functions/awaitMessage.js (hunk 1 above)
Comment (kyranet) [reply functions/awaitMessage.js]: Updated.

**Review by CyberiumShadow — APPROVED** [review]

**Pull request merged**